src/counter.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::time::{Duration, UNIX_EPOCH, SystemTime};
+
+pub struct Counter {
+    pub name: &'static str,
```

Comment:
Could that be String?

Reply:
we are using a static, I think `str` is more appropriate.